File: easy/majority_frequency_group.py
def majority_frequency_group(s: str) -> str:
    entries, freq = {}, {}
    for char in s:
        if char in freq:
            freq[char] += 1
        else:
            freq[char] = 1
    for k, v in freq.items():
        if v in entries:
            entries[v] += k
        else:
            entries[v] = k

    groups = [
        (v, k)
        for k, v in sorted(entries.items(), key=lambda x: (-len(x[1]), -x[0]))
    ]
    return groups[0][0]

File: easy/MajorityFrequencyGroup.cs
namespace Easy;

/// <summary>
///
/// </summary>
public class MajorityFrequencyGroup
{
    /// <summary>
    ///
    /// </summary>
    public static string Solve(string s)
    {
        Dictionary<char, int> freq = [];
        Dictionary<int, string> entries = [];
        foreach (char c in s)
        {
            freq.TryGetValue(c, out int count);
            freq[c] = count + 1;
        }
        foreach ((char key, int value) in freq)
        {
            entries[value] = entries.GetValueOrDefault(value, "") + key;
        }
        List<KeyValuePair<int, string>> groups = [.. entries];
        groups.Sort(
            (lhs, rhs) =>
            {
                int cmp = rhs.Value.Length.CompareTo(lhs.Value.Length);
                if (cmp != 0)
                    return cmp;
                return rhs.Key.CompareTo(lhs.Key);
            }
        );
        return groups.First().Value;
    }
}
